pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+					return
+				}
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expect.deleteResult, result)
+				assert.Equal(t, tt.expect.completed, completed)
+			}
+		})
+	}
+}
+
+func TestDelegateError_Get(t *testing.T) {
+	type input struct {
+		ctx     context.Context
+		name    string
+		options *metav1.GetOptions
+	}
+
+	type output struct {
+		getResult runtime.Object
+		err       error
+	}
+
+	type testCase struct {
+		name       string
+		storeSetup func(*MockStore[*TestType, *TestTypeList])
+		wantErr    bool
+		input      input
+		expect     output
+	}
+
+	tests := []testCase{
+		{
+			name: "working case",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name:    "test-object",
+				options: &metav1.GetOptions{},
+			},
+			expect: output{
+				getResult: &TestType{},
+				err:       nil,
+			},
+			wantErr: false,
+		},
+		{
+			name:       "missing user in the context",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {},
+			input: input{
+				ctx:     context.Background(),
+				name:    "testing-obj",
+				options: &metav1.GetOptions{},
+			},
+			expect: output{
+				getResult: nil,
+				err:       errMissingUserInfo,
+			},
+			wantErr: true,
+		},
+		{
+			name: "store get error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, assert.AnError)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name:    "test-object",
+				options: &metav1.GetOptions{},
+			},
+			expect: output{
+				getResult: &TestType{},
+				err:       assert.AnError,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			addToSchemeTest(scheme)
+
+			gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+			gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+			tt.storeSetup(mockStore)
+
+			testDelegate := &delegateError[*TestType, *TestTypeList]{
+				inner: &delegate[*TestType, *TestTypeList]{
+					scheme: scheme,
+					t:      &TestType{},
+					tList:  &TestTypeList{},
+					gvk:    gvk,
+					gvr:    gvr,
+					store:  mockStore,
+				},
+			}
+
+			result, err := testDelegate.Get(tt.input.ctx, tt.input.name, tt.input.options)
+			if tt.wantErr {
+				// check if we have an error
+				assert.Error(t, err)
+
+				// check if error is an apierror
+				if _, ok := err.(apierrors.APIStatus); ok {
+					assert.True(t, ok)
+					return
+				}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.